mllm/models/qwen_npu/tokenization_qwen.hpp

```
@@ -0,0 +1,331 @@
+  explicit QwenTokenizer(const std::string& vocab_file, const std::string& merge_file, bool split_special_tokens = false)
+      : split_special_tokens_(split_special_tokens) {
+    preprocessor::initLocal();
+    preprocessor::makeBytes2UnicodeMap(bytes_2_unicode_dict_);
+    for (auto& kv : bytes_2_unicode_dict_) { bytes_2_unicode_dict_inverse_.insert({kv.second, kv.first}); }
+
+    bpe_.initFromSentencePieceJson(vocab_file);
+    initBpeRanks(merge_file);
+    initSpecialTokens();
+  }
+
+  void initBpeRanks(const std::string& merge_file) {
```

Clang (14.0.6)

[error] method 'initBpeRanks' can be made static

(readability-convert-member-functions-to-static,-warnings-as-errors)